sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponse.java

```diff
@@ -209,4 +210,24 @@ public String getFaultInjectionRuleId() {
     public void setFaultInjectionRuleId(String faultInjectionRuleId) {
         this.faultInjectionRuleId = faultInjectionRuleId;
     }
+
+    public StoreResponse withRemappedStatusCode(int newStatusCode, double additionalRequestCharge) {
+
+        Map<String, String> headers = new HashMap<>();
+        for (int i = 0; i < this.responseHeaderNames.length; i++) {
+            String headerName = this.responseHeaderNames[i];
+            if (headerName.equalsIgnoreCase(HttpConstants.HttpHeaders.REQUEST_CHARGE)) {
+                double currentRequestCharge = this.getRequestCharge();
+                double newRequestCharge = currentRequestCharge + additionalRequestCharge;
+                headers.put(headerName, String.valueOf(newRequestCharge));
+            } else {
+                headers.put(headerName, this.responseHeaderValues[i]);
+            }
+        }
+
+        return new StoreResponse(
```

Comment:
Why are we creating a new `StoreResponse` object here? We can use the existing one as creating new references to the byte[] content may creating unnecessary JVM objects

Reply:
Would lead to weird situation when debugging - another ResourceResponse or CosmosItemResposne with different status code and headers that the underlying StoreResponse. I think it is safer to clone when changing the status code and headers. Overhead of instantiating JVM object compared to the cost accrued for I/O int his case (do the additional read) is negligible.